### ISSUE TITLE
Unexpected needless_range_loop, expected explicit_counter_loop

### ISSUE THREAD

**Issue by vallentin** [rust-lang/rust-clippy#4472]
The following code triggered the [needless_range_loop][needless_range_loop] lint, which would be a false positive, given the condition uses `i`.

```rust
let arr = [10, 20, 30];

for i in 0..arr.len() {
    if i != 1 {
        println!("{}", arr[i]);
    }
}
```

Instead the [explicit_counter_loop][explicit_counter_loop] lint would be expected.

```rust
let arr = [10, 20, 30];

for (i, x) in arr.iter().enumerate() {
    if i != 1 {
        println!("{}", x);
    }
}
```

Version: clippy 0.0.212 (082cfa7 2019-06-07)

[needless_range_loop]: https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
[explicit_counter_loop]: https://rust-lang.github.io/rust-clippy/master/index.html#explicit_counter_loop

**Comment by flip1995** [comment]
Oh, this is in fact the correct lint. The documentation of the `explicit_counter_loop` is just wrong / misleading.

https://github.com/rust-lang/rust-clippy/blob/ba6681300e9d7a10379c9fbbf539f4f3261fdc6a/clippy_lints/src/loops.rs#L1426-L1433

https://github.com/rust-lang/rust-clippy/blob/ba6681300e9d7a10379c9fbbf539f4f3261fdc6a/clippy_lints/src/loops.rs#L1442-L1443

So the `explicit_counter_loop` lint should just trigger on contructs like
```rust
let i = 0;
for x in vec {
    // ...
    i += 1;
}
```
